src/test/java/seedu/address/logic/parser/CheckoutCommandParserTest.java

```diff
@@ -28,19 +29,30 @@
 
     @Test
     public void parse_validArgs_returnsCheckoutCommand() {
-        assertParseSuccess(parser, ROOM_DESC_001, new CheckoutCommand(TypicalRoomNumbers.ROOM_NUMBER_001));
+        RoomNumber expectedRoomNumber = new RoomNumber(VALID_ROOM_NUMBER_001);
+        BookingPeriod expectedBookingPeriod = new BookingPeriod(VALID_DATE_START_AMY, VALID_DATE_END_AMY);
+        assertParseSuccess(parser, ROOM_DESC_001, new CheckoutCommand(expectedRoomNumber));
         assertParseSuccess(parser, ROOM_DESC_001 + DATE_START_DESC_AMY + DATE_END_DESC_AMY,
-                new CheckoutCommand(TypicalRoomNumbers.ROOM_NUMBER_001,
-                        new BookingPeriod(VALID_DATE_START_AMY, VALID_DATE_END_AMY)));
+                new CheckoutCommand(expectedRoomNumber, expectedBookingPeriod));
     }
 
     @Test
     public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "", String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
         assertParseFailure(parser, "1", String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, "01", String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
         assertParseFailure(parser, "001", String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, "-r 001",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
         assertParseFailure(parser, ROOM_DESC_001 + DATE_START_DESC_AMY,
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, ROOM_DESC_001 + DATE_END_DESC_AMY,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, DATE_START_DESC_AMY + DATE_END_DESC_AMY,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
         assertParseFailure(parser, " r/001 from/01/01/18 to/30/02/18",
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, CheckoutCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, " r/001 from/2018/01/01 to/2018/01/02",
```

Comment:
Since this parse failure is on the date format, the `ROOM_DESC_001` can still be used.

Reply:
okay will change so it makes it clearer which is the part that is supposed to cause the failure